realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -456,7 +456,7 @@ public String toJson() {
      */
     public boolean isValid() {
         Token userToken = getSyncUser().getUserToken();
-        return syncUser.isLoggedIn() && userToken != null && userToken.expiresMs() > System.currentTimeMillis();
+        return syncUser.isLoggedIn() && SyncManager.getUserStore().isActive(syncUser.getIdentity()) && userToken != null && userToken.expiresMs() > System.currentTimeMillis();
```

Comment:
no this just checks if the ObjectStore user is marked for logout or not. OS doesn't check token validity